Sources/Plasma/PubUtilLib/plNetGameLib/Private/plNglAuth.cpp

```diff
@@ -2387,6 +2388,20 @@ static bool Recv_ScoreGetHighScoresReply(
     return true;
 }
 
+//============================================================================
+static bool Recv_ServerCaps(
+    const uint8_t   msg[],
+    unsigned        bytes,
+    void *          param
+    ) {
+    const Auth2Cli_ServerCaps & caps = *(const Auth2Cli_ServerCaps *)msg;
+
+    hsReadOnlyStream stream(caps.byteCount, reinterpret_cast<const void*>(caps.buffer));
+    ((CliAuConn*)param)->caps.Read(&stream);
```

Comment:
Note that this technically has the size twice, since `hsBitVector::Read` also includes a 32-bit dword count.  I don't see any reasonable way around this though, and this message should only be transferred once per client connection.